### PR TITLE
Canopy Year: nlcd_tcc to the EE Catalog

### PR DESCRIPTION
**Pull request by ajgriebel**
working on jsonnet, example js, and preview js files; coordinating with Josh at GTAC on information. @joshheyer 
1 band (image) per year within an image collection, values are 0-100%, 255 is the background (water).

### CODE REVIEW

**Comment by schwehr** [comment]
Looking at this file, it appears that the data should be restructured.

Inspecting with this:

```
var image = ee.ImageCollection('projects/usgs-gee-audra-griebel/assets/eros/canopy/year');
print(image);
```

I see each image has a different band name. The asset id already has the year:

```
projects/usgs-gee-audra-griebel/assets/eros/canopy/year/2011
...
projects/usgs-gee-audra-griebel/assets/eros/canopy/year/2021
```

I suggest naming the band `cover`

Also, I would suggest setting nodata value was not specified in the source files so there is a large region around the data with a value of 255 just doesn't appear.

**Comment by schwehr** [comment]
Also, each year should have the start and end data set for the year, I'm seeing them as the same:

Image projects/usgs-gee-audra-griebel/assets/eros/canopy/year/2011 has:

```
system: time_end: 1703980800000
system:time_start: 1672531200000
```

and

Image projects/usgs-gee-audra-griebel/assets/eros/canopy/year/2021 has:

```
system:time_end: 1703980800000
system:time_start: 1672531200000
```

File: catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    multiple channels and platforms. 
+
+    For more information visit [Multi-Resolution Land Characteristics Consortium](https://www.mrlc.gov/data)
+    or [The USFS Tree Canopy Cover Datasets page](https://data.fs.usda.gov/geodata/rastergateway/treecanopycover/).
+  |||,
+  license: license.id,
+  links: ee.standardLinks(subdir, id),
+  keywords: [
+    'landcover',
+    'tree-cover',
```
Comment (schwehr) [inline catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet:66]: `tree-cover` → `tree_cover`

File: catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  ],
+  id: id,
+  title: 'Tree Canopy Cover (TCC) data (2011-2021)' + version,
+  version: 'version',
+
+  // TODO(schwehr): what if it is an image collection
+  'gee:type': ee_const.gee_type.imagecollection,
+
+  description: |||
+    The USDA Forest Service (USFS) builds two versions of percent tree canopy cover data,
```
Comment (schwehr) [inline catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet:37]: Fit to 80 columns

**Review by schwehr — CHANGES_REQUESTED** [review]
Missing the example and preview scripts.

Take a look at https://github.com/google/earthengine-catalog/pull/640 for me taking a pass at the file.

**Comment by ajgriebel** [comment]
I will re-run these to update the band names to canopy. (FYI there will be more images coming, he wants me to include the Science TCC (tree canopy cover) and SE (standard error) products (which will be another 28 images). I'm having Adam and Patrick help me get them uploaded to the bucket while I'm out). 
I also will work on the jsonnet file as best as possible. 
I'll get the example and previews in there eventually.
It's always challenging getting work done on vacation. Thanks for your patience and help Kurt!

File: catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    The non-processing area is represented by value 254, and the background is represented by the value 255. 
+    The Science and NLCD tree canopy cover data are accessible for multiple user communities, through 
+    multiple channels and platforms. 
+
+    For more information visit [Multi-Resolution Land Characteristics Consortium](https://www.mrlc.gov/data)
+    or [The USFS Tree Canopy Cover Datasets page](https://data.fs.usda.gov/geodata/rastergateway/treecanopycover/).
+  |||,
+  license: license.id,
+  links: ee.standardLinks(subdir, id),
+  keywords: [
```
[the remark this answers is not in the document]
Comment (ajgriebel) [reply catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet:70]: Updated

File: catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    'tree-cover',
+    'usgs',
+    'gtac',
+    'usfs',
+    'yearly',
+    'nlcd',
+    'mrlc',
+  ],
+  providers: [
+    ee.producer_provider(
```
[the remark this answers is not in the document]
Comment (ajgriebel) [reply catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet:81]: Updated

File: catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet (hunk 2 above)
Comment (ajgriebel) [reply catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet]: I can't ever see the column width. Is there a button to show that? 
I believe I got this fixed (I copied the raw from the test file so it should be right).

File: catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+                    '2011-01-01T00:00:00Z', '2022-01-01T00:00:00Z'),
+  'gee:interval': {
+    type: 'cadence',
+    unit: 'year',
+    interval: 1,
+  },
+  summaries: {
+    gsd: [30],
+    'eo:bands': [
+      {
```
[the remark this answers is not in the document]
Comment (ajgriebel) [reply catalog/USGS/USGS_NLCD_RELEASES_2021_REL_CANOPY_YEAR.jsonnet]: I'm having a hard time understanding the bands (why this one only has one? wouldn't each year be a band? or no? sorry, trying to understand!)

**Comment by ajgriebel** [comment]
@simonff would you please be able to take a look at this? The asset path needs some tweaking. Image Collection location is here: https://code.earthengine.google.com/?asset=projects/usgs-gee-audra-griebel/assets/eros/canopy/nlcd_tcc_year. When I try running the js and preview js it pulls this error: year_2021 %: Layer error: ImageCollection.load: ImageCollection asset 'USGS/NLCD_RELEASES/2021_REL/CANOPY/NLCD/TCC/YEAR' not found (does not exist or caller does not have access). Also, for some reason the CI isn't on the checker anymore (it appears it used to be). Thank you!

**Comment by ajgriebel** [comment]
Josh Heyer will be handling the pull request for the Canopy and including it with all of their GTAC Canopy information.

**Pull request closed without merging**